nc/source/physics/IntersectionQueries.cpp

```diff
@@ -39,65 +69,81 @@ namespace nc::physics
         return squareDistance <= sphere.radius * sphere.radius;
     }
 
-    bool Gjk(const BoundingVolume& a, const BoundingVolume& b, FXMMATRIX aMatrix, FXMMATRIX bMatrix)
+    bool Gjk(const BoundingVolume& a, const BoundingVolume& b, FXMMATRIX aMatrix, FXMMATRIX bMatrix, CollisionState* stateOut)
     {
-        auto aRotation = GetRotation(aMatrix);
-        auto bRotation = GetRotation(bMatrix);
+        IF_THROW(!stateOut, "Gjk - CollisionState cannot be null");
 
+        stateOut->simplex = Simplex{};
+        stateOut->rotationA = GetRotation(aMatrix);
+        stateOut->rotationB = GetRotation(bMatrix);
         Vector3 direction = Vector3::One();
-        Simplex simplex;
         size_t itCount = 0;
-        constexpr size_t maxIterations = 10u;
 
-        while(++itCount <= maxIterations)
+        while(++itCount <= GjkMaxIterations)
         {
             const auto direction_v = XMLoadVector3(&direction);
-            auto aDirection_v = XMVector3InverseRotate(direction_v, aRotation);
+            auto aDirection_v = XMVector3InverseRotate(direction_v, stateOut->rotationA);
             auto aSupport_v = XMVector3Transform(MinkowskiSupport(a, aDirection_v), aMatrix);
-            auto bDirection_v = XMVector3InverseRotate(XMVectorNegate(direction_v), bRotation);
+            auto bDirection_v = XMVector3InverseRotate(XMVectorNegate(direction_v), stateOut->rotationB);
             auto bSupport_v = XMVector3Transform(MinkowskiSupport(b, bDirection_v), bMatrix);
             auto support_v = aSupport_v - bSupport_v;
 
             if(XMVector3Less(XMVector3Dot(support_v, direction_v), g_XMZero))
-                return false;
+                break;
 
             Vector3 support;
             XMStoreVector3(&support, support_v);
 
-            simplex.push_front(support);
+            stateOut->simplex.push_front(support);
 
-            if(RefineSimplex[simplex.size() - 1](simplex, direction))
+            if(RefineSimplex[stateOut->simplex.size() - 1](stateOut->simplex, direction))
                 return true;
         }
 
         return false;
     }
 
-    bool Gjk(const BoundingVolume& a, const BoundingVolume& b)
+    NormalData Epa(const BoundingVolume& a, const BoundingVolume& b, DirectX::FXMMATRIX aMatrix, DirectX::FXMMATRIX bMatrix, CollisionState* state)
     {
-        Vector3 direction = Vector3::One();
-        Simplex simplex;
-        size_t itCount = 0;
-        constexpr size_t maxIterations = 10u;
+        state->polytope.Initialize(state->simplex);
+        auto minFace = state->polytope.ComputeNormalData();
+        NormalData minNorm{Vector3::Zero(), FloatMax};
```

Comment:
Is this MinkowskiNormal or MinimumNormal? If the former will you call it like minkNorm for clarity?